src/main/java/com/faforever/client/preferences/ui/SettingsController.java

```diff
@@ -589,5 +602,30 @@ public void onPatchGamePrefsForMultipleInstances() {
       notificationService.addImmediateErrorNotification(e, "settings.fa.patchGamePrefsFailed");
     }
   }
+
+  public void onAddMirrorURL() {
+    String text = mirrorURITextField.getText();
+    if (text.isBlank()) {
+      return;
+    }
+    if (!text.endsWith("/")) {
+      text = text + "/";
+    }
+
+    URI uri = null;
+    try {
+      uri = new URL(text).toURI();
+    } catch (URISyntaxException | MalformedURLException e) {
+      log.debug("Failed to add invalid URL: {}", text, e);
+      return;
+    }
+
+    if (mirrorURLsListView.getItems().contains(uri)) {
+      return;
+    }
+    preferencesService.getPreferences().getMirror().getMirrorURLs().add(uri);
+    preferencesService.storeInBackground();
+    mirrorURITextField.clear();
```

Comment:
Since you have a return in the catch anyway everything after the catch can just be moved into the try statement, this removes the possibility of uri being null as well since you can move the declaration there as well.